packages/module/src/tracking/segment_tracking_provider.ts

```diff
@@ -0,0 +1,58 @@
+import { AnalyticsBrowser } from '@segment/analytics-next';
+
+import { TrackingApi, TrackingEventProperties } from './tracking_api';
+import { InitProps, TrackingSpi } from './tracking_spi';
+
+export class SegmentTrackingProvider implements TrackingSpi, TrackingApi {
+  private analytics: AnalyticsBrowser | undefined;
+  getKey(): string {
+    return 'segmentKey';
+  }
+
+  initialize(props: InitProps): void {
+    // eslint-disable-next-line no-console
+    console.log('SegmentProvider initialize');
+    const segmentKey = props.segmentKey as string;
+
+    this.analytics = AnalyticsBrowser.load(
+      {
+        writeKey: segmentKey,
+        cdnURL: props.segmentCdn as string
+      },
+
+      {
+        integrations: {
+          ...props.segmentIntegrations
```

Comment:
Do you know what the purpose of this integrations object is? I cannot figure out what values would be expected and valid here. 

You can only use `...` before a variable or object if it is actually an iterable type like an array or json object. if it's a primitive like a string, then you cannot use the `...` spreading operator.

Reply:
It would be a json object.

This would be an example of the `InitProps`:

 ```
segmentIntegrations: {
          'Segment.io': {
            apiHost: 'my.proxy.host/connections/api/v1',
            protocol: 'https'
          }
        },
```